ui/components/app/update-modal/update-modal.tsx

```diff
@@ -70,18 +70,14 @@ function UpdateModal() {
         </ModalBody>
         <ModalFooter
           onSubmit={async () => {
-            try {
-              await browser.tabs.create({
-                url: 'https://metamask.io/updating',
-                active: true,
-              });
-            } catch (error) {
-              console.error(error);
-            }
-            await requestSafeReload();
+            setIsLoading(true);
+            await openUpdateTabAndReload();
+            setIsLoading(false);
```

Comment:
Not requesting a change here, since I'd much rather ship this sooner, rather than polish an error experience no one is likely to ever see! But its good to know:

The general pattern for loading -> await doThingForAWhile() -> notLoading is to do the "notLoading" part in a `finally` block, and add error handling in the `catch` block if the `await doThingForAWhile()` step rejects.

Again, I don't think you should make changes here to this (as it was technically already a "problem"). Just something I noticed and thought i'd mention.